sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -146,12 +148,26 @@ case class ScriptTransformationExec(
           }
         }
 
+        private def waitForTransformationProcessTermination(): Unit = {
+          val timeout = SQLConf.get.getConf(SQLConf.TRANSFORMATION_EXIT_TIMEOUT)
+          try {
+            proc.waitFor(timeout, TimeUnit.SECONDS)
+          } catch {
+            case t: Throwable =>
+              log.warn(s"Transformation script process exits timeout in ${timeout} seconds", t)
```

Comment:
```
     * @param timeout the maximum time to wait
     * @param unit the time unit of the {@code timeout} argument
     * @return {@code true} if the subprocess has exited and {@code false} if
     *         the waiting time elapsed before the subprocess has exited.
     * @throws InterruptedException if the current thread is interrupted
     *         while waiting.
     * @throws NullPointerException if unit is null
```

It seems that timeout won't throw an exception but only return false?

Reply:
Yes, it is bad mistake. Code updated.